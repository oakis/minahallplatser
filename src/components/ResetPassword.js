import React, { Component } from 'react';
import { View, TouchableWithoutFeedback, Keyboard } from 'react-native';
import { connect } from 'react-redux';
import { emailChanged, resetUserPassword, resetRoute } from '../actions';
import { Input, Button, Message } from './common';

class ResetPassword extends Component {

	componentWillMount() {
		this.props.resetRoute();
	}

	componentWillUnmount() {
		this.props.resetRoute();
	}

	onEmailChange(text) {
		this.props.emailChanged(text);
	}

	onButtonPress() {
		this.props.loading = true;
		const { email } = this.props;
		this.props.resetUserPassword(email);
	}

	render() {
		return (
			<TouchableWithoutFeedback
				onPress={() => Keyboard.dismiss()}
			>
				<View
					style={{
						flex: 1,
						marginLeft: 10,
						marginRight: 10,
						flexDirection: 'column',
						justifyContent: 'center',
						alignItems: 'center'
					}}
				>
					<Message
						type="danger"
						message={this.props.error}
					/>
					<Input
						placeholder="din@email.se"
						keyboardType="email-address"
						returnKeyType="next"
						onChangeText={this.onEmailChange.bind(this)}
						value={this.props.email}
						icon="ios-mail"
					/>
					<Button
						loading={this.props.loading}
						uppercase
						color="primary"
						label="Återställ lösenord"
						onPress={this.onButtonPress.bind(this)}
					/>
				</View>
			</TouchableWithoutFeedback>
		);
	}

}

const mapStateToProps = ({ auth, errors }) => {
	const { email, loading } = auth;
	const { error } = errors;
	return { email, error, loading };
};

export default connect(mapStateToProps,
	{ emailChanged, resetUserPassword, resetRoute }
)(ResetPassword);
