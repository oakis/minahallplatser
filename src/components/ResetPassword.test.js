import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ResetPassword from './ResetPassword';


const mockStore = configureMockStore([thunk]);

const initialState = {
    auth: {
        email: 'abc@123.com',
        loading: false,
    },
    errors: {
        error: 'An error message',
    },
};

it('should match snapshot', () => {
    const wrapper = shallow(<ResetPassword store={mockStore(initialState)} resetRoute={jest.fn()} />).dive();
    expect(wrapper).toMatchSnapshot();
});
