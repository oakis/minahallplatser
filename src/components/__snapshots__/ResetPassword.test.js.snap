// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResetPassword
    email="abc@123.com"
    error="An error message"
    loading={false}
    resetRoute={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "marginLeft": 10,
            "marginRight": 10,
          }
        }
      >
        <Message
          message="An error message"
          type="danger"
        />
        <Input
          icon="ios-mail"
          keyboardType="email-address"
          onChangeText={[Function]}
          placeholder="din@email.se"
          returnKeyType="next"
          value="abc@123.com"
        />
        <Button
          color="primary"
          label="Återställ lösenord"
          loading={false}
          onPress={[Function]}
          uppercase={true}
        />
      </View>,
      "onPress": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Message
            message="An error message"
            type="danger"
          />,
          <Input
            icon="ios-mail"
            keyboardType="email-address"
            onChangeText={[Function]}
            placeholder="din@email.se"
            returnKeyType="next"
            value="abc@123.com"
          />,
          <Button
            color="primary"
            label="Återställ lösenord"
            loading={false}
            onPress={[Function]}
            uppercase={true}
          />,
        ],
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 10,
          "marginRight": 10,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "message": "An error message",
            "type": "danger",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": "ios-mail",
            "keyboardType": "email-address",
            "onChangeText": [Function],
            "placeholder": "din@email.se",
            "returnKeyType": "next",
            "value": "abc@123.com",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "color": "primary",
            "label": "Återställ lösenord",
            "loading": false,
            "onPress": [Function],
            "uppercase": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
              "marginLeft": 10,
              "marginRight": 10,
            }
          }
        >
          <Message
            message="An error message"
            type="danger"
          />
          <Input
            icon="ios-mail"
            keyboardType="email-address"
            onChangeText={[Function]}
            placeholder="din@email.se"
            returnKeyType="next"
            value="abc@123.com"
          />
          <Button
            color="primary"
            label="Återställ lösenord"
            loading={false}
            onPress={[Function]}
            uppercase={true}
          />
        </View>,
        "onPress": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Message
              message="An error message"
              type="danger"
            />,
            <Input
              icon="ios-mail"
              keyboardType="email-address"
              onChangeText={[Function]}
              placeholder="din@email.se"
              returnKeyType="next"
              value="abc@123.com"
            />,
            <Button
              color="primary"
              label="Återställ lösenord"
              loading={false}
              onPress={[Function]}
              uppercase={true}
            />,
          ],
          "style": Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "marginLeft": 10,
            "marginRight": 10,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "message": "An error message",
              "type": "danger",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "icon": "ios-mail",
              "keyboardType": "email-address",
              "onChangeText": [Function],
              "placeholder": "din@email.se",
              "returnKeyType": "next",
              "value": "abc@123.com",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "color": "primary",
              "label": "Återställ lösenord",
              "loading": false,
              "onPress": [Function],
              "uppercase": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
